traits/tests/test_trait_list_object.py

```diff
@@ -0,0 +1,315 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+import copy
+import pickle
+import unittest
+
+from traits.trait_errors import TraitError
+from traits.trait_list_object import adapt_trait_validator, TraitList
+from traits.trait_types import _validate_int
+
+
+def int_validator(trait_list, index, removed, added):
+    if isinstance(index, slice):
+        return [_validate_int(item) for item in added]
+    else:
+        if isinstance(added, list):
+            return [_validate_int(item) for item in added]
+        else:
+            return _validate_int(added)
+
+
+def list_validator(trait_list, index, removed, added):
+    if isinstance(added, list):
```

Comment:
This doesn't look right. Shouldn't we be verifying that every element of `added` is a list?